src/utils/atomWithStorage.ts

```diff
@@ -0,0 +1,31 @@
+import { atom, WritableAtom } from 'jotai'
+import type { SetStateAction } from '../core/types'
+
+export function atomWithStorage<Value>(key: string, initialValue: Value) {
+  type Update = SetStateAction<Value>
```

Comment:
Do we accept function update? If so, L24-L26 should be implemented so. There's no magic.

Reply:
Can you elaborate?

Reply:
It has to be:
```ts
    (get, set, update: Update) => {
      const newValue = typeof update === 'function' ? update(get(baseAtom)) : update
      set(baseAtom, newValue)
      window.localStorage.setItem(key, JSON.stringify(newValue))
    }
```